.pre-commit-config.yaml

```diff
@@ -0,0 +1,36 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: 'v3.4.0'
+    hooks:
+      - id: end-of-file-fixer
+        exclude_types: [svg]
+      - id: trailing-whitespace
+        exclude_types: [svg]
+      - id: check-json
+      - id: check-yaml
+      - id: check-case-conflict
+  - repo: https://github.com/google/yapf
+    rev: 'v0.31.0'
+    hooks:
+      - id: yapf
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: '3.9.1'
+    hooks:
+      - id: flake8
+        args:
+        - --show-source
+        additional_dependencies:
+          - pep8-naming
+          - flake8-docstrings
+          - flake8-rst-docstrings
+  - repo: local
+    hooks:
+      - id: clang-format
+        name: clang-format
+        entry: clang-format -i
+        additional_dependencies: [-c conda-forge, clang-tools=10.0]
```

Comment:
Looks like 11.1 is the current version. https://anaconda.org/conda-forge/clang-tools

Reply:
As you pointed out to me, one of the advantages to pre-commit is versioned tools. When I ran clang-format with the system language on my system, clang-format 11 reformatted a lot of files as it handles a few things differently. I set it to this more to prove that it could match the existing clang10 formatted code.

I should probably update to the latest version.

Reply:
Should we put this in a repository so we can reuse the `clang-format` configuration in the `repo` list?

Reply:
If you want, sure. It could be used for all our C++ projects.

Reply:
Along the same idea, would you want to try the pre-commit checks with a more minimal Docker container? Like “miniconda3” or something similar? IIRC there are no specific requirements for the glotzerlab software container.

Reply:
I'm currently running it in the bare GitHub Actions VM. I could switch this to run just the clang-format checks and use pre-commit.ci for the Python language checks.

Reply:
done: https://github.com/glotzerlab/pre-commit-clang-format